sql/expression_resolver_xgb.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sql
+
+import (
+	"fmt"
+	"strconv"
+)
+
+type resolvedXGBTrainClause struct {
+	NumBoostRound int
+	Maximize      bool
+	ParamsAttr    map[string]*attribute
+}
+
+func resolveXGBTrainClause(tc *trainClause) (*resolvedXGBTrainClause, error) {
```

Comment:
try put attribute extraction into the codegen for each submitter?

Reply:
It's used for mapping [XGBoost Parameters](https://xgboost.readthedocs.io/en/latest/parameter.html)

Reply:
If we only have these two parameters currently, try all put it in `expression_resolver.go` and refine this later. We should let each codegen to deal with it's only attributes.